gneiss/regression/_ols.py

```diff
@@ -407,37 +431,27 @@ def lovo(self, **kwargs):
            R2diff : np.array, float
                Decrease in Rsquared for each variable left out.
         """
-        endog = self.balances
-        exog_names = self.results[0].model.exog_names
-        exog = pd.DataFrame(self.results[0].model.exog,
-                            index=self.balances.index,
-                            columns=exog_names)
-        cv_iter = LeaveOneOut(len(exog_names))
-        results = pd.DataFrame(index=exog_names,
+        cv_iter = LeaveOneOut(len(self.design_matrix.columns))
+        results = pd.DataFrame(index=self.design_matrix.columns,
                                columns=['mse', 'Rsquared', 'R2diff'],
                                dtype=np.float64)
         _r2 = self.r2
         for i, (inidx, outidx) in enumerate(cv_iter):
-            feature_id = exog_names[i]
-
-            model_i = _fit_ols(y=endog, x=exog.loc[:, inidx], **kwargs)
-            res_i = OLSModel(model_i, balances=endog)
-            res_i.fit()
-
-            # See `statsmodels.regression.linear_model.RegressionResults`
-            # for more explanation on `ess` and `ssr`.
-            # sum of squares regression.
-            ssr = sum([r.ess for r in res_i.results])
-            # sum of squares error.
-            sse = sum([r.ssr for r in res_i.results])
-            # calculate the overall coefficient of determination (i.e. R2)
-            sst = sse + ssr
-            r2_left_out = 1 - sse / sst
-            # degrees of freedom for residuals
-            dfe = res_i.results[0].df_resid
-            results.loc[feature_id, 'mse'] = sse / dfe
-            results.loc[feature_id, 'Rsquared'] = r2_left_out
-            results.loc[feature_id, 'R2diff'] = _r2 - r2_left_out
+
+            feature_id = self.design_matrix.columns[i]
+
+            res_i = OLSModel(Y=self.response_matrix,
+                             Xs=self.design_matrix.iloc[:, inidx])
+            res_i.fit(**kwargs)
+            predicted = res_i.predict()
+            r = self.response_matrix.values
+            p = predicted.values
+
+            model_resid = ((r - p)**2)
+            model_mse = np.mean(model_resid.sum(axis=0))
+            results.loc[feature_id, 'mse'] = model_mse
+            results.loc[feature_id, 'Rsquared'] = res_i.r2
+            results.loc[feature_id, 'R2diff'] = self.r2 - res_i.r2
```

Comment:
This is trying to do some relative importance analysis - basically trying to see how much the R^2 decreases when a specific variable is left out.  @serenejiang do you have any thoughts here?